modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordSerializer.java

```diff
@@ -0,0 +1,186 @@
+    /** Table ID. */
+    private final IgniteUuid tableId;
+
+    /** Mapper. */
+    private final Mapper<R> mapper;
+
+    /** Simple mapping mode: single column maps to a basic type. For example, {@code RecordView<String>}.  */
+    private final boolean oneColumnMode;
+
+    /**
+     * Constructor.
+     *
+     * @param tableId       Table ID.
+     * @param mapper        Mapper.
+     */
+    public ClientRecordSerializer(IgniteUuid tableId, Mapper<R> mapper) {
+        assert tableId != null;
+        assert mapper != null;
+
+        this.tableId = tableId;
+        this.mapper = mapper;
+
+        oneColumnMode = MarshallerUtil.mode(mapper.targetType()) != null;
+    }
+
+    public Mapper<R> mapper() {
+        return mapper;
+    }
+
+    public void writeRec(@Nullable  R rec, ClientSchema schema, ClientMessagePacker out, TuplePart part) {
+        out.packIgniteUuid(tableId);
+        out.packInt(schema.version());
+
+        writeRecRaw(rec, schema, out, part);
+    }
+
+    public void writeRecRaw(@Nullable R rec, ClientSchema schema, ClientMessagePacker out, TuplePart part) {
+        Marshaller marshaller = schema.getMarshaller(mapper, part);
+        ClientMarshallerWriter writer = new ClientMarshallerWriter(out);
+
+        try {
+            marshaller.writeObject(rec, writer);
+        } catch (MarshallerException e) {
+            throw new IgniteClientException(e.getMessage(), e);
+        }
+    }
+
+    public void writeRecs(@Nullable R rec, @Nullable R rec2, ClientSchema schema, ClientMessagePacker out, TuplePart part) {
+        out.packIgniteUuid(tableId);
+        out.packInt(schema.version());
+
+        Marshaller marshaller = schema.getMarshaller(mapper, part);
+        ClientMarshallerWriter writer = new ClientMarshallerWriter(out);
+
+        try {
+            marshaller.writeObject(rec, writer);
+            marshaller.writeObject(rec2, writer);
+        } catch (MarshallerException e) {
+            throw new IgniteClientException(e.getMessage(), e);
+        }
+    }
+
+    public void writeRecs(@NotNull Collection<R> recs, ClientSchema schema, ClientMessagePacker out, TuplePart part) {
+        out.packIgniteUuid(tableId);
+        out.packInt(schema.version());
+        out.packInt(recs.size());
+
+        Marshaller marshaller = schema.getMarshaller(mapper, part);
+        ClientMarshallerWriter writer = new ClientMarshallerWriter(out);
+
+        try {
+            for (R rec : recs) {
+                marshaller.writeObject(rec, writer);
+            }
+        } catch (MarshallerException e) {
+            throw new IgniteClientException(e.getMessage(), e);
+        }
+    }
+
+    public Collection<R> readRecsNullable(ClientSchema schema, ClientMessageUnpacker in) {
+        return readRecs(schema, in, true, TuplePart.KEY_AND_VAL);
+    }
+
+    public Collection<R> readRecs(ClientSchema schema, ClientMessageUnpacker in) {
+        return readRecs(schema, in, false, TuplePart.KEY_AND_VAL);
+    }
```

Comment:
Let's drop if these methods are not used.

Reply:
Fixed.